---
title: "pyscenic_import_r"
output: html_document
date: "2024-01-23"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

##Importing the results from pyscenic 
```{r}
pyScenicDir <- "./data/week30/pyscenic"
library(SCENIC)
library(tidyverse)
```

#Reading the loom file 
```{r}
library(SCopeLoomR)
pyScenicLoomFile <- file.path(pyScenicDir,"output.loom")
loom <- open_loom(pyScenicLoomFile, mode = "r")
```

#Read info from the loom file 
```{r}
# Read information from loom file:
regulons_incidMat <- get_regulons(loom, column.attr.name='Regulons')
#A list with tf and its target genes 
regulons <- regulonsToGeneLists(regulons_incidMat)
#Regulon activity across the cells 
regulonAUC <- get_regulons_AUC(loom, column.attr.name='RegulonsAUC')
regulonsAucThresholds <- get_regulon_thresholds(loom)
#embeddings <- get_embeddings(loom)
```



```{r}
#Get the expression matrix that is used to compute AUC 
exprMat <- get_dgem(loom)
#cellInfo <- get_cell_annotation(loom)
#clusterings <- get_clusterings_with_name(loom)
```




```{r}
cellInfo <- readRDS(file = "./data/week30/rda/cellInfo.rds")
cellInfo$seurat_clusters_ind <- paste(cellInfo$genotype,cellInfo$seurat_clusters,
                                      sep = "_")
```


```{r}
library(AUCell)
library(grid)
cellInfo <- data.frame(cellInfo)
#regulonAUC <- loadInt(scenicOptions, "aucell_regulonAUC")
regulonAUC <- regulonAUC[onlyNonDuplicatedExtended(rownames(regulonAUC)),]

regulonActivity_byCellType <- sapply(split(rownames(cellInfo), cellInfo$genotype_clusters),
                                     function(cells) rowMeans(getAUC(regulonAUC)[,cells]))

regulonActivity_byCellType_Scaled <- t(scale(t(regulonActivity_byCellType), center = T, scale=T))

ComplexHeatmap::Heatmap(regulonActivity_byCellType, 
                        name="Regulon activity",
                        row_names_gp = grid::gpar(fontsize = 7))
#Be careful about scaling and centering the data

ComplexHeatmap::Heatmap(regulonActivity_byCellType_Scaled, 
                        name="Regulon activity",
                        row_names_gp = grid::gpar(fontsize = 8),
                        column_names_gp = grid::gpar(fontsize = 7)
                        )
```




#Function to subset on specific cell type and compare between "WT" vs "DNMT3A" and "TET2"
```{r}
Find_diff_TFs <- function(celltype) {
  
  #celltypes <- cellInfo$seurat_clusters
  
  #Have this huge matrix with all cells vs the 266 TFs at an individual level 
  regulonActivity_by_indcells <- getAUC((regulonAUC))
  tfs <- names(regulons)
  
  #Subset for celltype 
  cellInfo_celltype <- cellInfo[cellInfo$seurat_clusters == celltype,]
  selected_cols <- intersect(colnames(regulonActivity_by_indcells),rownames(cellInfo_celltype))
  regulonActivity_celltype <- regulonActivity_by_indcells[,selected_cols] %>%
                              t()

  combined_ann_tf <- merge(as.data.frame(regulonActivity_celltype), 
                          cellInfo_celltype, by = "row.names", 
                          all.x = TRUE)
  
  tf_wt_tet2 <- combined_ann_tf %>%
               filter(genotype == "wt_45_1" | 
               genotype == "tet2_45_2")

  wt_data <- tf_wt_tet2 %>%
             filter(genotype == "wt_45_1") %>%
             select(all_of(tfs))
  
  tet2_data <- tf_wt_tet2 %>%
               filter(genotype == "tet2_45_2") %>%
               select(all_of(tfs))
  
  diffexptf_wt_tet2 <- c()
  log2_fc_tf_tet2 <- c()
  for (tf in tfs) {
    p_val_tf <- wilcox.test(wt_data[,tf],tet2_data[,tf],paired = FALSE)$p.value
    diffexptf_wt_tet2[tf] <- p_val_tf
    median_wt <- median(wt_data[,tf])
    median_tet2 <- median(tet2_data[,tf])
    comp_median <- (median_tet2/median_wt)
    foldchange_auc <- log2((abs(comp_median))^(sign(comp_median)))
    log2_fc_tf_tet2[tf] <- foldchange_auc
    }
  
  diffexptf_wt_tet2 <- diffexptf_wt_tet2 %>% 
                       as.data.frame() %>%
                       rename({{ celltype }} := ".")
  
  log2_fc_tf_tet2 <- log2_fc_tf_tet2 %>% 
                       as.data.frame() %>%
                       rename({{ celltype }} := ".")
  
  ##Now the comparission for wt vs dnmt3a 
  tf_wt_dnmt3a <- combined_ann_tf %>%
               filter(genotype == "wt_45_1" | 
               genotype == "dnmt3a_45_2")
  
  wt_data <- tf_wt_dnmt3a %>%
             filter(genotype == "wt_45_1") %>%
             select(all_of(tfs))
  
  dnmt3a_data <- tf_wt_dnmt3a %>%
               filter(genotype == "dnmt3a_45_2") %>%
               select(all_of(tfs))

  diffexptf_wt_dnmt3a <- c()
  log2_fc_tf_dnmt3a <- c()
  for (tf in tfs) {
    p_val_tf <- wilcox.test(wt_data[,tf],dnmt3a_data[,tf],paired = FALSE)$p.value
    median_wt <- median(wt_data[,tf])
    median_dnmt3a <- median(dnmt3a_data[,tf])
    comp_median <- (median_dnmt3a/median_wt)
    foldchange_auc <- log2((abs(comp_median))^(sign(comp_median)))
    diffexptf_wt_dnmt3a[tf] <- p_val_tf
    log2_fc_tf_dnmt3a[tf] <- foldchange_auc
    }
  diffexptf_wt_dnmt3a <- diffexptf_wt_dnmt3a %>% 
                         as.data.frame() %>%
                         rename({{ celltype }} := ".")
  
   log2_fc_tf_dnmt3a <- log2_fc_tf_dnmt3a %>% 
                       as.data.frame() %>%
                       rename({{ celltype }} := ".")
  
  return(list(diffexptf_wt_tet2 = diffexptf_wt_tet2,log2_fc_tf_tet2 = log2_fc_tf_tet2, 
              diffexptf_wt_dnmt3a = diffexptf_wt_dnmt3a,log2_fc_tf_dnmt3a=log2_fc_tf_dnmt3a))
}
```

#Find significant p-vals
```{r}
#Note to use the -log10 transformed matrix 
sig_Tfs <- function(df, threshold = 1.3) {
  sig_tfs <- list()
  for (col_name in colnames(df)) {
    # Get the row names where the value is greater than the threshold for each column
    selected_rows <- rownames(df)[df[, col_name] > threshold]
    # Store the result in the list
    sig_tfs[[col_name]] <- selected_rows
  }
  return(result_list)
  }
```




```{r}
celltypes <- levels(cellInfo$seurat_clusters) 

#In order to run this function : regulonAUC, regulons and tfs must be defined 

p_val_all <- lapply(celltypes,Find_diff_TFs) #Warnings are for vascular fibroblasts 
names(p_val_all) <- celltypes

```


```{r}
# Extract all first data frames from each pair which are comparissions between 
#wt vs tet2
wt_tet2_pval <- lapply(p_val_all, function(pair) pair[[1]])

# Use do.call and cbind to bind all first data frames together
wt_tet2_pval <- do.call(cbind, wt_tet2_pval) %>% 
                as.data.frame()
```


```{r}
wt_tet2_fc <- lapply(p_val_all, function(pair) pair[[2]])
wt_tet2_fc <- do.call(cbind, wt_tet2_fc) %>% 
                as.data.frame()

#Only for resident like macrophages between wt vs tet2 : 
df_macs <- cbind(wt_tet2_pval$`Resident-like Macrophages`,
                 wt_tet2_fc$`Resident-like Macrophages`) %>%
            as.data.frame() %>%
            rename(p_val = "V1",
                   log2fc = "V2") %>%
            mutate(p_adj = p.adjust(p_val,method = "BH")) %>%
            mutate(y= -log10(p_adj)) %>%
            mutate(y = pmin(y, 20))


df_macs$log2fc <- ifelse(df_macs$log2fc > 0.25, 0.25,
                  ifelse(df_macs$log2fc < -0.25, -0.25,
                         df_macs$log2fc))


row.names(df_macs) <- rownames(wt_tet2_pval)


# add a column of NAs
df_macs$diffexpressed <- "NO"
# if log2Foldchange > 0.6 and pvalue < 0.05, set as "UP" 
df_macs$diffexpressed[df_macs$log2fc > 0.05 & df_macs$y >= 10] <- "UP"
# if log2Foldchange < -0.6 and pvalue < 0.05, set as "DOWN"
df_macs$diffexpressed[df_macs$log2fc < -0.05 & df_macs$y >= 10] <- "DOWN"



# Create a new column "delabel" to de, that will contain the name of genes differentially expressed (NA in case they are not)
df_macs$delabel <- NA
df_macs$delabel[df_macs$diffexpressed != "NO"] <- rownames(df_macs[df_macs$diffexpressed != "NO",])




ggplot(data=df_macs, aes(x=log2fc, y= y, col=diffexpressed,
                         label = delabel))+ 
    geom_point() + 
    theme_minimal()+
    geom_text()


```












```{r}
small_constant <- 1e-300
wt_tet2_pval_log <- wt_tet2_pval %>%
                    mutate_all(~ -log10(. + small_constant))

wt_tet2_pval_log_scale <- scale(wt_tet2_pval_log, center = T, scale=T)
```




```{r}
# Extract all second data frames from each pair which are comparissions between 
#wt vs dnmt3a
wt_dnmt3a_pval <- lapply(p_val_all, function(pair) pair[[2]])

# Use do.call and cbind to bind all second data frames together
wt_dnmt3a_pval <- do.call(cbind, wt_dnmt3a_pval)
```


```{r}
p_val_wt_tet2_scale <- t(scale(t(wt_tet2_pval), center = T, scale=T))

ComplexHeatmap::Heatmap(wt_tet2_pval, 
                        name="p-val : wt vs tet2",
                        row_names_gp = grid::gpar(fontsize = 7))
#Be careful about scaling and centering the data

ComplexHeatmap::Heatmap(p_val_wt_tet2_scale, 
                        name="log10(p-value) : wt vs tet2",
                        #row_names_gp = grid::gpar(fontsize = 8),
                        column_names_gp = grid::gpar(fontsize = 7)
                        )
```


```{r}
p_val_wt_tet2_scale %>%
  as.data.frame() %>%
  ggplot()+
  aes(x = p_val_wt_tet2_scale$`Resident-like Macrophages`) +
  geom_histogram(bins = 50)
```



```{r}
p_val_wt_tet2_scale <-  p_val_wt_tet2_scale %>% as.data.frame()
tfs_res_sig <- p_val_wt_tet2_scale[p_val_wt_tet2_scale$`Resident-like Macrophages` <= -2,]
```









##Write another function to plot TF distribution of AUC across the conditions 
#WT vs DNMT3A or WT vs TET2






#WT vs TET2
```{r}
mean_Jun <- combined_ann_Jun %>%
         filter(mutation == "wt" | 
         mutation == "tet2") %>%
         group_by(mutation) %>%
         summarize(mean = mean(Jun))

combined_ann_Jun %>%
  filter(mutation == "wt" | 
         mutation == "tet2") %>%
  ggplot(aes(x = Jun, fill = mutation)) +
  #geom_histogram(aes(y = ..density..), alpha = 0.5, position = "identity", bins = 30) +
  geom_density(alpha = 0.5) +
  geom_vline(data = mean_Jun, aes(xintercept = mean, color = mutation),
             linetype = "dashed")+
  labs(title = "Density Plot by wt vs tet2", x = "Jun(+)", y = "Density") +
  theme_minimal()
  #scale_fill_viridis_d()
 
```






#Find target genes of Jun between Wt and tet2 
```{r}
target_genes_Jun <- regulons$`Jun(+)`
rdaDir <- ("./data/week30/rda/")
load(file = paste0(rdaDir,"merged_seurat_harmony.rda"))
library(Seurat)
features = target_genes_Jun[100:111]
res_macs_seurat <- subset(merged_seurat_harmony,
                          seurat_clusters == "Resident-like Macrophages" &
                          (genotype =="wt_45_1"| genotype =="tet2_45_2"))

VlnPlot(res_macs_seurat, features = features,split.by = "genotype")


```



```{r}
VlnPlot(res_macs_seurat, features = "Selenbp1",split.by = "genotype")
VlnPlot(res_macs_seurat, features = "Egr2",split.by = "genotype")
VlnPlot(res_macs_seurat, features = "Klf2",split.by = "genotype")
VlnPlot(res_macs_seurat, features = "Lyl1",split.by = "genotype")
VlnPlot(res_macs_seurat, features = "Maf",split.by = "genotype")
VlnPlot(res_macs_seurat, features = "Egr2",split.by = "genotype")
VlnPlot(res_macs_seurat, features = "Irf1",split.by = "genotype")
VlnPlot(res_macs_seurat, features = "Xbp1",split.by = "genotype")
VlnPlot(res_macs_seurat, features = "Ifi47",split.by = "genotype")
VlnPlot(res_macs_seurat, features = "Fam46a",split.by = "genotype")
VlnPlot(res_macs_seurat, features = "Klf2",split.by = "genotype")
```




























##Subset on cell types and compare wild type vs mutant  
```{r}
#Subsetting on macrophages 

#Fisrt subset only on the raw values form the regulon activity as scaling is done based on other cell types too 

#First only on "Inflmatory Macrophages"
columns_to_keep <- grep("Inflamatory Macrophages", colnames(regulonActivity_byCellType), value = TRUE)

# Subset the data frame to keep only the selected columns
regulonActivity_InflamatoryMacs <- regulonActivity_byCellType[, columns_to_keep]

var_InflamatoryMacs <- apply(regulonActivity_InflamatoryMacs,1,var)
# Sort rows based on variances
sorted_rows <- order(var_InflamatoryMacs, decreasing = TRUE)
# Select the top 50 rows
top_50_InflamtoryMacs <- regulonActivity_InflamatoryMacs[sorted_rows[1:50], ]

#First lets make a heatmap to get an idea about the regulons that are most likely to be changed 
#Then we can plot histograms with the distribution of their AUC activity 

#Scale now within the celltype 
top_50_InflamtoryMacs_scaled <- t(scale(t(top_50_InflamtoryMacs), center = T, scale=T))

ComplexHeatmap::Heatmap(top_50_InflamtoryMacs, 
                        name="Regulon activity",
                        row_names_gp = grid::gpar(fontsize = 8),
                        column_names_gp = grid::gpar(fontsize = 6))

```


```{r}
#Scaled version of the heatmap 

ComplexHeatmap::Heatmap(top_50_InflamtoryMacs_scaled, 
                        name="Regulon activity",
                        row_names_gp = grid::gpar(fontsize = 8),
                        column_names_gp = grid::gpar(fontsize = 7)
                        )
```


#We need to get the regulonActivity again : This time by individual cell types and samples : So 5 in total

```{r}
regulonActivity_byCellType_ind <- sapply(split(rownames(cellInfo), cellInfo$seurat_clusters_ind),
                                     function(cells) rowMeans(getAUC(regulonAUC)[,cells]))
```



##Subset on cell types and compare wild type vs mutant(Differnece being we are doing on an individual sample scale)
```{r}
#Subsetting on macrophages 

#Fisrt subset only on the raw values form the regulon activity as scaling is done based on other cell types too 

#First only on "Inflmatory Macrophages"
columns_to_keep <- grep("Inflamatory Macrophages", colnames(regulonActivity_byCellType_ind), value = TRUE)

# Subset the data frame to keep only the selected columns
regulonActivity_InflamatoryMacs_ind <- regulonActivity_byCellType_ind[, columns_to_keep]

var_InflamatoryMacs_ind <- apply(regulonActivity_InflamatoryMacs_ind,1,var)
# Sort rows based on variances
sorted_rows <- order(var_InflamatoryMacs_ind, decreasing = TRUE)
# Select the top 50 rows
top_50_InflamtoryMacs_ind <- regulonActivity_InflamatoryMacs_ind[sorted_rows[1:50], ]

#First lets make a heatmap to get an idea about the regulons that are most likely to be changed 
#Then we can plot histograms with the distribution of their AUC activity 

#Scale now within the celltype 
top_50_InflamtoryMacs_scaled_ind <- t(scale(t(top_50_InflamtoryMacs_ind), center = T, scale=T))

ComplexHeatmap::Heatmap(top_50_InflamtoryMacs_ind, 
                        name="Regulon activity",
                        row_names_gp = grid::gpar(fontsize = 8),
                        column_names_gp = grid::gpar(fontsize = 6))

```


```{r}
#Scaled version of the heatmap 

ComplexHeatmap::Heatmap(top_50_InflamtoryMacs_scaled_ind, 
                        name="Regulon activity",
                        row_names_gp = grid::gpar(fontsize = 8),
                        column_names_gp = grid::gpar(fontsize = 7)
                        )
```

##Do the exact same thing on Resident-like Macrophages 

```{r}
#Now on "Resident-like Macrophages"
columns_to_keep <- grep("Resident-like Macrophages", colnames(regulonActivity_byCellType), value = TRUE)

# Subset the data frame to keep only the selected columns
regulonActivity_resMacs <- regulonActivity_byCellType[, columns_to_keep]

var_resMacs <- apply(regulonActivity_resMacs,1,var)
# Sort rows based on variances
sorted_rows <- order(var_resMacs, decreasing = TRUE)
# Select the top 50 rows
top_50_resMacs <- regulonActivity_resMacs[sorted_rows[1:50], ]

#First lets make a heatmap to get an idea about the regulons that are most likely to be changed 
#Then we can plot histograms with the distribution of their AUC activity 

#Scale now within the celltype 
top_50_resMacs_scaled <- t(scale(t(top_50_resMacs), center = T, scale=T))

ComplexHeatmap::Heatmap(top_50_resMacs, 
                        name="Regulon activity",
                        row_names_gp = grid::gpar(fontsize = 8),
                        column_names_gp = grid::gpar(fontsize = 6))

```


```{r}
#Scaled version of the heatmap for resident-like macrophages

ComplexHeatmap::Heatmap(top_50_resMacs_scaled, 
                        name="Regulon activity",
                        row_names_gp = grid::gpar(fontsize = 8),
                        column_names_gp = grid::gpar(fontsize = 7)
                        )
```

##Resident like macrophages at the individual level 

```{r}

#on "Resident-like Macrophages"
columns_to_keep <- grep("Resident-like Macrophages", colnames(regulonActivity_byCellType_ind), value = TRUE)

# Subset the data frame to keep only the selected columns
regulonActivity_resMacs_ind <- regulonActivity_byCellType_ind[, columns_to_keep]

var_resMacs_ind <- apply(regulonActivity_resMacs_ind,1,var)
# Sort rows based on variances
sorted_rows <- order(var_resMacs_ind, decreasing = TRUE)
# Select the top 50 rows
top_50_resMacs_ind <- regulonActivity_resMacs_ind[sorted_rows[1:50], ]

#First lets make a heatmap to get an idea about the regulons that are most likely to be changed 
#Then we can plot histograms with the distribution of their AUC activity 

#Scale now within the celltype 
top_50_resMacs_scaled_ind <- t(scale(t(top_50_resMacs_ind), center = T, scale=T))

ComplexHeatmap::Heatmap(top_50_resMacs_ind, 
                        name="Regulon activity",
                        row_names_gp = grid::gpar(fontsize = 8),
                        column_names_gp = grid::gpar(fontsize = 6))

```



#Scaled version for resident like macs 

```{r}
#Scaled version of the heatmap 

ComplexHeatmap::Heatmap(top_50_resMacs_scaled_ind, 
                        name="Regulon activity",
                        row_names_gp = grid::gpar(fontsize = 8),
                        column_names_gp = grid::gpar(fontsize = 7)
                        )
```

#Write a function to get the AUC values, plot the heatmap for ind vs all together, so that dont have to copy paste so many times 




##Plot histogram of AUC values for a few tfs 
```{r}
#First step is to get the regulon activity of individual cells 

#Have this huge matrix with all cells vs the 266 TFs at an individual level 
regulonActivity_by_indcells <- getAUC((regulonAUC))

#Subset for resident macrophages 
cellInfo_resMacs <- cellInfo[cellInfo$seurat_clusters == "Resident-like Macrophages",]
selected_cols <- intersect(colnames(regulonActivity_by_indcells),rownames(cellInfo_resMacs))
regulonActivity_resMacs_ind_cells <- regulonActivity_by_indcells[,selected_cols]

#Can select tf for example : Here we select Jun 
regulonActivity_by_indcells_Jun <- regulonActivity_resMacs_ind_cells %>%
                                   t() %>%
                                   as.data.frame() %>%
                                   select("Jun(+)") %>%
                                   rename(Jun = "Jun(+)")

combined_ann_Jun <- merge(as.data.frame(regulonActivity_by_indcells_Jun), 
                          cellInfo_resMacs, by = "row.names", 
                          all.x = TRUE)
library(ggplot2)


ggplot(combined_ann_Jun, aes(x = Jun, fill = genotype)) +
  geom_histogram(aes(y = ..density..), alpha = 0.5, position = "identity", bins = 30) +
  geom_density(alpha = 0.6) +
  labs(title = "Histogram and Density Plot by Sample", x = "Jun(+)", y = "Density") +
  theme_minimal()
  
     
```

#WT vs TET2
```{r}
mean_Jun <- combined_ann_Jun %>%
         filter(mutation == "wt" | 
         mutation == "tet2") %>%
         group_by(mutation) %>%
         summarize(mean = mean(Jun))

combined_ann_Jun %>%
  filter(mutation == "wt" | 
         mutation == "tet2") %>%
  ggplot(aes(x = Jun, fill = mutation)) +
  #geom_histogram(aes(y = ..density..), alpha = 0.5, position = "identity", bins = 30) +
  geom_density(alpha = 0.5) +
  geom_vline(data = mean_Jun, aes(xintercept = mean, color = mutation),
             linetype = "dashed")+
  labs(title = "Density Plot by wt vs tet2", x = "Jun(+)", y = "Density") +
  theme_minimal()
  #scale_fill_viridis_d()
 
```

#Only one wt vs tet2 
```{r}
mean_Jun <- combined_ann_Jun %>%
         filter(genotype == "wt_45_1" | 
         genotype == "tet2_45_2") %>%
         group_by(genotype) %>%
         summarize(mean = mean(Jun))

combined_ann_Jun %>%
  filter(genotype == "wt_45_1" | 
         genotype == "tet2_45_2") %>%
  ggplot(aes(x = Jun, fill = genotype)) +
  #geom_histogram(aes(y = ..density..), alpha = 0.5, position = "identity", bins = 30) +
  geom_density(alpha = 0.5) +
  geom_vline(data = mean_Jun, aes(xintercept = mean, color = genotype),
             linetype = "dashed")+
  labs(title = "Density Plot by wt vs tet2", x = "Jun(+)", y = "Density") +
  theme_minimal()
  #scale_fill_viridis_d()
 
```



#WT vs TET2
```{r}
mean_Jun <- combined_ann_Jun %>%
         filter(mutation == "wt" | 
         mutation == "tet2") %>%
         group_by(genotype) %>%
         summarize(mean = mean(Jun))

combined_ann_Jun %>%
  filter(mutation == "wt" | 
         mutation == "tet2") %>%
  ggplot(aes(x = Jun, fill = genotype)) +
  #geom_histogram(aes(y = ..density..), alpha = 0.5, position = "identity", bins = 30) +
  geom_density(alpha = 0.5) +
  geom_vline(data = mean_Jun, aes(xintercept = mean, color = genotype),
             linetype = "dashed")+
  labs(title = "Density Plot by wt vs tet2", x = "Jun(+)", y = "Density") +
  theme_minimal() +
  scale_fill_viridis_d()

```


#WT vs DNMT3A
```{r}
mean_Jun <- combined_ann_Jun %>%
         filter(mutation == "wt" | 
         mutation == "dnmt3a") %>%
         group_by(mutation) %>%
         summarize(mean = mean(Jun))

combined_ann_Jun %>%
  filter(mutation == "wt" | 
         mutation == "dnmt3a") %>%
  ggplot(aes(x = Jun, fill = mutation)) +
  #geom_histogram(aes(y = ..density..), alpha = 0.5, position = "identity", bins = 30) +
  geom_density(alpha = 0.5) +
  geom_vline(data = mean_Jun, aes(xintercept = mean, color = mutation),
             linetype = "dashed")+
  labs(title = "Density Plot by wt vs dnmt3a", x = "Jun(+)", y = "Density") +
  theme_minimal()
  #scale_fill_viridis_d()
```


#WT vs DNMT3A
```{r}
mean_Jun <- combined_ann_Jun %>%
         filter(mutation == "wt" | 
         mutation == "dnmt3a") %>%
         group_by(genotype) %>%
         summarize(mean = mean(Jun))

combined_ann_Jun %>%
  filter(mutation == "wt" | 
         mutation == "dnmt3a") %>%
  ggplot(aes(x = Jun, fill = genotype)) +
  #geom_histogram(aes(y = ..density..), alpha = 0.5, position = "identity", bins = 30) +
  geom_density(alpha = 0.5) +
  geom_vline(data = mean_Jun, aes(xintercept = mean, color = genotype),
             linetype = "dashed")+
  labs(title = "Density Plot by wt vs dnmt3a", x = "Jun(+)", y = "Density") +
  theme_minimal() +
  scale_fill_viridis_d()
```

#Wilcoxon test : WT VS TET2
```{r}
jun_wt_tet2 <- combined_ann_Jun %>%
               filter(mutation == "wt" | 
               mutation == "tet2")

wt_data <- jun_wt_tet2 %>%
           filter(mutation == "wt") %>%
           select(Jun)
tet2_data <- jun_wt_tet2 %>%
           filter(mutation == "tet2") %>%
           select(Jun)

shapiro.test(wt_data$Jun)
shapiro.test(tet2_data$Jun)

wilcox.test(wt_data$Jun,tet2_data$Jun,paired = FALSE)
wilcox.test(tet2_data$Jun,wt_data$Jun,paired = FALSE)
```


#Wilcoxon test : WT vs DNMT3A
```{r}
jun_wt_dnmt3a <- combined_ann_Jun %>%
               filter(mutation == "wt" | 
               mutation == "dnmt3a")

wt_data <- jun_wt_dnmt3a %>%
           filter(mutation == "wt") %>%
           select(Jun)

dnmt3a_data <- jun_wt_dnmt3a %>%
           filter(mutation == "dnmt3a") %>%
           select(Jun)

shapiro.test(wt_data$Jun)
shapiro.test(dnmt3a_data$Jun)

wilcox.test(wt_data$Jun,dnmt3a_data$Jun,paired = FALSE)
wilcox.test(dnmt3a_data$Jun,wt_data$Jun,paired = FALSE)
```



```{r}
ggplot(combined_ann_Jun, aes(x = Jun, fill = mutation)) +
  geom_histogram(aes(y = ..density..), alpha = 0.5, position = "identity", bins = 30) +
  geom_density(alpha = 0.5) +
  labs(title = "Histogram and Density Plot by Genotype", x = "Jun(+)", y = "Density") +
  theme_minimal()+
  scale_fill_viridis_d()
```


##Compute RSS score from regulon matrix 


```{r}
rss <- calcRSS(AUC=getAUC(regulonAUC), cellAnnotation=cellInfo[colnames(regulonAUC), "genotype_clusters"])
rssPlot <- plotRSS(rss)
plotly::ggplotly(rssPlot$plot)
```




##Compute RSS scores on cell types and not all combined 
#First lets try for resident-like macrophages 

```{r}

#Subset for resident macrophages 
cellInfo_resMacs <- cellInfo[cellInfo$seurat_clusters == "Resident-like Macrophages",]
selected_cols <- intersect(colnames(regulonActivity_by_indcells),rownames(cellInfo_resMacs))
regulonActivity_resMacs_ind_cells <- regulonActivity_by_indcells[,selected_cols]
rss_res_macs <- calcRSS(AUC=regulonActivity_resMacs_ind_cells,
                        cellAnnotation=cellInfo_resMacs[colnames(regulonActivity_resMacs_ind_cells), 
                                                        "mutation"])

var_rss_resMacs_ind <- apply(rss_res_macs,1,var)
# Sort rows based on variances
sorted_rows <- order(var_rss_resMacs_ind, decreasing = TRUE)
# Select the top 50 rows
top_50_resMacs_ind_rss <- rss_res_macs[sorted_rows[1:75], ]

rssPlot <- plotRSS(top_50_resMacs_ind_rss)
plotly::ggplotly(rssPlot$plot)

```



##To visualize embeddings/trajectories calculated using other methods :

```{r}
library(Seurat)
rdaDir <- ("/Users/xsankt/Desktop/proj1_tet_dnmt3a_mouse/data/week30/rda/")
load(file = paste0(rdaDir,"merged_seurat_harmony.rda"))
dr_coords <- Embeddings(merged_seurat_harmony, reduction="umap")

###Embedings in Seurat are also calulated only on the highly variable genes. 
tfs <- c("Jun(+)")
par(mfrow=c(2,2))
AUCell::AUCell_plotTSNE(dr_coords, cellsAUC=selectRegulons(regulonAUC, tfs), plots=c("AUC", "histogram","binary"), cex = .5)
```



```{r}
DimPlot(merged_seurat_harmony, reduction = "umap", label = T,label.size = 2)
```

##Have to find a way to set the thresholds for binarized version of the regulon activity 

```{r}

```



##Find top regulators 
```{r}

```



#Resident-like Macrophages 
```{r}
res_macs_fc <- p_val_all[["Resident-like Macrophages"]][["log2_fc_tf_dnmt3a"]][["Resident-like Macrophages"]]%>% as.matrix()
```


```{r}
wt_data[] <- lapply(wt_data, function(col) {
  min_after_zero <- min(col[which(col > 0)])
  col[col == 0] <- min_after_zero
  return(col)
})
```



